Migrate to  phpspec/prophecy-phpunit

athos-ribeiro opened the pull request:
https://github.com/sebastianbergmann/phpunit/issues/5033 changed phpunit
to no longer depend on phpspec/prophecy. This results in the need to add
a development dependency for phpspec/prophecy. Doing so results in a
warning, since PHPUnit\Framework\TestCase::prophesize() is deprecated
and will be removed in PHPUnit 10. Let's use the trait provided by
phpspec/prophecy-phpunit instead.

* Fixes #177

athos-ribeiro committed:
Migrate to  phpspec/prophecy-phpunit

https://github.com/sebastianbergmann/phpunit/issues/5033 changed phpunit
to no longer depend on phpspec/prophecy. This results in the need to add
a development dependency for phpspec/prophecy. Doing so results in a
warning, since PHPUnit\Framework\TestCase::prophesize() is deprecated
and will be removed in PHPUnit 10. Let's use the trait provided by
phpspec/prophecy-phpunit instead.

* Fixes #177

athos-ribeiro commented:
Note that phpspec/prophecy-phpunit v2.0.0 requires php ^7.3. One alternative here would be to just require phpspec/prophecy instead. But then, such support will be dropped for phpunit 10.

Thoughts?

mnapoli commented:
Thanks for the PR!

I would be fine dropping support for older PHP versions and require PHP 7.4+, would that work?

athos-ribeiro commented:
It works for me, but then, would it cause any trouble for other users? If in doubt we can always  just require phpspec/prophecy  and delay the decision!

mnapoli commented:
> If in doubt we can always just require phpspec/prophecy and delay the decision!

Ah sorry I missed that, yeah that sounds great!

athos-ribeiro committed:
Revert "Migrate to  phpspec/prophecy-phpunit"

This reverts commit e03ad061e153680feb575e8da9195f68af67a7f4.
Supporting phpspec/prophecy-phpunit at this point, would require
dropping php <= 7.2 support. We can re-visit this matter once we start
supporting phpunit 10.

athos-ribeiro committed:
Require phpspec/prophecy as a dev dependency

While e03ad06 (reverted) is not applied, we require phpspec/prophecy as
a development dependency. This is needed because phpunit dropped
ptophecy as a dependency in recent releases. See e03ad06 for further
reference.

athos-ribeiro commented:
I lowered the dependency version to one that should accept php 7.1 now :)

szepeviktor commented:
Done in #183

mnapoli commented:
🚀
